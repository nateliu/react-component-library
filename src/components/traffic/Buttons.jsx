import React, { Component } from 'react';
import { GO, CAUTION, STOP } from '../../constants/traffic/actionTypes';
import './Buttons.less';

const Buttons = ({
    go,
    caution,
    stop,
    lightStatus
}) => {
    return (
        <div style={{ textAlign: 'center' }}>
            <button className='trafficButton' onClick={go}
                disabled={lightStatus == GO || lightStatus == CAUTION}>
                Go
            </button>

            <button className='trafficButton' onClick={caution}
                disabled={lightStatus == CAUTION || lightStatus == STOP}>
                Caution
            </button>

            <button className='trafficButton' onClick={stop}
                disabled={lightStatus == STOP || lightStatus == GO}>
                Stop
            </button>
        </div>
    )
}

export default Buttons;