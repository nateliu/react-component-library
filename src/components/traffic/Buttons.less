  .trafficButton {
    height: 30px;
    width: 80px;
    border-radius: 4px;
    margin-left: 10px;
    cursor: 'pointer'
  }