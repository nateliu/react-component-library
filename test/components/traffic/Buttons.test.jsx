import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Buttons from '../../../src/components/traffic/Buttons';

function setup() {
    const props = {
        go :jest.fn(),
        caution:jest.fn(),
        stop: jest.fn(),
        lightStatus:"GO"
    }

    const enzymeWrapper = shallow(<Buttons {...props} />)

    return {
        props,
        enzymeWrapper
    }
}

describe('traffic/Buttons', () => {
    it('renders correctly with default props', () => {
        const { enzymeWrapper } = shallow(<Buttons />)
        expect(shallowToJson(enzymeWrapper)).toMatchSnapshot();
    });

    it('renders correctly with GO Status', () => {
        const { enzymeWrapper } = setup()
        expect(enzymeWrapper.find('.trafficButton').length).toEqual(3);
        expect(shallowToJson(enzymeWrapper)).toMatchSnapshot();
    })
})