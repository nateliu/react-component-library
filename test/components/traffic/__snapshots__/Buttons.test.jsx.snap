// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`traffic/Buttons renders correctly with GO Status 1`] = `
<div
  style={
    Object {
      "textAlign": "center",
    }
  }
>
  <button
    className="trafficButton"
    disabled={true}
    onClick={[MockFunction]}
  >
    Go
  </button>
  <button
    className="trafficButton"
    disabled={false}
    onClick={[MockFunction]}
  >
    Caution
  </button>
  <button
    className="trafficButton"
    disabled={true}
    onClick={[MockFunction]}
  >
    Stop
  </button>
</div>
`;

exports[`traffic/Buttons renders correctly with default props 1`] = `null`;
